src/OrchardCore.Modules/OrchardCore.AdminMenu/Controllers/NodeController.cs

```diff
@@ -344,5 +352,27 @@ public async Task<IActionResult> Toggle(string id, string treeNodeId)
 
             return Ok();
         }
+
+
```

Comment:
you don't need to make an ajax call at all. or limit the permissions returned. they're static (so it's not an open ended sql call), so just return them all from the driver in one hit.

Reply:
I agree on that too.